src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -190,17 +192,78 @@ public static int EncodeInPlace(Span<byte> buffer, int sourceLength)
             return encodedLength;
         }
 
+        #endregion
+
+        #region Decode
+
+        public static int ComputeDecodedLength(ReadOnlySpan<byte> source)
+        {
+            ref byte srcBytes = ref source.DangerousGetPinnableReference();
```

Comment:
It is hard to measure the perf for this scenario.
Depending on which test gets run first I am get conflicting results.

![image](https://cloud.githubusercontent.com/assets/6527137/26334801/7e109600-3f1c-11e7-876d-78f924c053be.png)

However, since we only have to index twice, I agree that we should use the indexer.

```C#
        private const int InnerCount = 1000000;

        [Benchmark(InnerIterationCount = InnerCount)]
        [InlineData(100)]
        [InlineData(1000)]
        [InlineData(1000 * 1000)]
        [InlineData(1000 * 1000 * 50)]
        private static void MyTestDangerous(int numberOfBytes)
        {
            var source = new byte[numberOfBytes];
            InitalizeBytes(source.AsSpan());

            foreach (var iteration in Benchmark.Iterations)
            {
                using (iteration.StartMeasurement())
                {
                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                        Base64Encoder.ComputeDecodedLengthA(source);
                }
            }
        }

        [Benchmark(InnerIterationCount = InnerCount)]
        [InlineData(100)]
        [InlineData(1000)]
        [InlineData(1000 * 1000)]
        [InlineData(1000 * 1000 * 50)]
        private static void MyTestIndexer(int numberOfBytes)
        {
            var source = new byte[numberOfBytes];
            InitalizeBytes(source.AsSpan());

            foreach (var iteration in Benchmark.Iterations)
            {
                using (iteration.StartMeasurement())
                {
                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
                        Base64Encoder.ComputeDecodedLengthB(source);
                }
            }
        }
```